cpp/oneapi/dal/array.hpp

```diff
@@ -317,6 +335,7 @@ class array {
     /// to the ownership structure in the array instance and :literal:`other`.
     array<T> operator=(array<T>&& other) {
         swap(*this, other);
+        other.reset_data();
```

Comment:
If we don't reset `other` it leads to errors in user code. There is no control from compiler side in the following code snippet:
```c++
auto x = get_array();
foo(std::move(x));
bar(x);
```

The fragment perfectly compiles, compiler has no diagnostic for that, no compile error or warnings. So it's recommended to "reset" state of moved objects. If we do that and the `bar` has respective asserts, we could detect this error at least in runtime. Otherwise `bar` would lead to segfault.

This is common practice in STL, if you check `::data()`/`::get()` of the moved `std::vector` or `std::shared_ptr` they return `nullptr`.

Reply:
I agree that it would be less error-prone. But current behavior is what I was requested to do w/o any additional logic upon it. Your behavior contradicts the description of the method and the spec. At least, they should be updated too.

Reply:
The new behavior is error-prone, but old behavior is not. Without reset, you don't even have chance to detect error.